### ISSUE TITLE
Kinase benchmark protocol

### ISSUE THREAD

**Issue by andrrizzi** [choderalab/kinase-benchmark#5]
Should we start planning how to setup the benchmark? In particular, I think we still need to decide:
- From which Abl and Src conformations/constructs should we start the calculations (i.e. **what RCSB ID**)?
- **Which subset of FDA approved inhibitors** do we want to run (if we are going to run all combinations for all 29 inhibitors it's going to take way more than a month)?
- We also need to **check whether the inhibitors we choose are prodrugs** (and maybe check stereochemistry?).
- How do we want to determine the **tautomeric/protonation state of the inhibitors**? Simply the ones with the lowest penalty in Epik?
- Which solvent models? We talked about 3 GB modes and 1 explicit solvent, so are we going to use **PME or RF for the explicit solvent**?

Technical question: how are we going to control the protonation/phosphorylation of Abl and Src? Can `PDBFixer` do that?

Tagging also @sonyahanson, @juliebehr and @steven-albanese .

**Comment by sonyahanson** [comment]
Possible Abl starting structures:
- PDB IDs 1IEP,1OPJ, 2HYY, 3K5V, 3MS9, 3MSS, and 3PYY are all Abl1-imatinib structures
- PDB ID 3CS9 is an Abl-nilotinib structure, and is pretty similar to the imatinib bound conformation.
- These should already be modeled on the cluster here:

`/cbio/jclab/projects/parton/kinome-ensembler/models/ABL1_HUMAN_D0/ABL1_*`

Possible Src starting structures:
- PDB IDs 2OIQ-chainA and 3OEZ are both Src-imatinib structures
- PDB ID 3EL7 is a Src-pyrazolopyrimidine3 structure, and is pretty similar to the imatinib bound conformation.
- These should already be modeled on the cluster here:

`/cbio/jclab/projects/parton/kinome-ensembler/models/SRC_HUMAN_D0/SRC_*`

If you want to check out the full list of Src and Abl DFG-OUT structures you can check here: https://github.com/choderalab/octomore/pull/14/files

**Comment by andrrizzi** [comment]
Awesome! Thanks!

**Comment by juliebehr** [comment]
@andrrizzi to your very last question, yes `PDBFixer` can do that!  (I guess it may remain to be seen if it's preferable, but it is capable.)

**Comment by juliebehr** [comment]
If it is of interest:
ABL1a: 1IEP, 2HYY, 3MS9, 3MSS, 3CS9(nilotinib)
ABL1b: 1OPJ, 3K5V, 3PYY

**Comment by sonyahanson** [comment]
Thanks @juliebehr ! Did you use a tool to find this out? It doesn't look like the sequence difference between these isoforms will matter much: http://www.uniprot.org/align/A2016021745V1TB8LFX  Also the models on the cluster will all be the same sequence, regardless of the template sequence (I think isoform A, but we should double check).

**Comment by jchodera** [comment]
Maybe we should add "isoform choice" to the list of potential sensitivity
investigations?

**Comment by andrrizzi** [comment]
> yes `PDBFixer` can do that

Perfect! By looking at the [manual](http://htmlpreview.github.io/?https://raw.github.com/pandegroup/pdbfixer/master/Manual.html) though, I can't figure out how to do it. Does it have something to do with `fixer.nonstandardResidues`?

**Comment by juliebehr** [comment]
Ah ok I think you're right; it depends on what you're trying to do with respect to protonation and phosphorylation.  Going through the steps related to `fixer.nonstandardResidues` will ensure you don't have any phosphorylated residues; it won't be helpful if you want to go in the other direction.  Also, protonation as controlled by `fixer.addMissingHydrogens` will only give you the ability to protonate based on default residue pKa and a specified pH.

```
    >>> fixer = PDBFixer(pdbid='2HYY')
    >>> fixer.findNonstandardResidues()
    >>> fixer.replaceNonstandardResidues()
    >>> fixer.addMissingHydrogens(pH=7.4)
```

If we want to go the other way around, there are parts of `perses` directly evolved from `PDBFixer` which notably use templates from `app.ForceField` instead of `PDBFixer` which includes template pdb files only for standard residues, and is therefore unable to modify residues to anything but these standard ones.

**Comment by andrrizzi** [comment]
> If we want to go the other way around, there are parts of `perses` directly evolved from `PDBFixer`

Cool! I'll take a look at them. Thanks!

**Comment by juliebehr** [comment]
It's basically the methods beginning [here](https://github.com/choderalab/perses/blob/master/perses/rjmc/topology_proposal.py#L311) and [here](https://github.com/choderalab/perses/blob/master/perses/rjmc/topology_proposal.py#L432) after having loaded templates [from `ForceField`](https://github.com/choderalab/perses/blob/master/perses/rjmc/topology_proposal.py#L554)  !

**Comment by andrrizzi** [comment]
Awesome! Thanks for the pointers!

**Comment by andrrizzi** [comment]
The protocol I'm running for the test. Let me know if you want me to try something else.

``` yaml
---
options:
  verbose: true
  mpi: yes
  output_dir: test_kinase
  temperature: 310*kelvin
  pressure: 1*atmosphere
  constraints: HBonds
  number_of_iterations: 200
  minimize: yes

molecules:
  abl:
    filepath: [2HYY_A.pdb, 3CS9_A.pdb]
    parameters: oldff/leaprc.ff99SBildn
    strip_protons: yes
  src:
    filepath: [2OIQ_A.pdb, 3EL7_A.pdb]
    parameters: oldff/leaprc.ff99SBildn
    strip_protons: yes
  imatinib:
    filepath: clinical-kinase-inhibitors.csv  # from the goodie-bag
    parameters: antechamber
    epik: 0
    select: 0  # select imatinib from csv
  bosutinib:
    filepath: clinical-kinase-inhibitors.csv
    parameters: antechamber
    epik: 0
    select: 3

solvents:
  RF:
    nonbonded_method: CutoffPeriodic
    nonbonded_cutoff: 1*nanometer
    clearance: 10*angstroms
    positive_ion: Na+
    negative_ion: Cl-
  GBSA:
    nonbonded_method: NoCutoff
    implicit_solvent: OBC2

protocols:
  absolute-binding:
    phases:
      complex:
        alchemical_path:
          lambda_electrostatics: [1.0, 0.97, 0.95, 0.90, 0.80, 0.70, 0.60, 0.50, 0.40, 0.30, 0.20, 0.10, 0.05, 0.025, 0.01, 0.0]
          lambda_sterics: [1.0, 0.97, 0.95, 0.90, 0.80, 0.70, 0.60, 0.50, 0.40, 0.30, 0.20, 0.10, 0.05, 0.025, 0.01, 0.0]
      solvent:
        alchemical_path:
          lambda_electrostatics: [1.0, 0.97, 0.95, 0.90, 0.80, 0.70, 0.60, 0.50, 0.40, 0.30, 0.20, 0.10, 0.05, 0.025, 0.01, 0.0]
          lambda_sterics: [1.0, 0.97, 0.95, 0.90, 0.80, 0.70, 0.60, 0.50, 0.40, 0.30, 0.20, 0.10, 0.05, 0.025, 0.01, 0.0]

experiments:
  components:
    receptor: [abl, src]
    ligand: [imatinib, bosutinib]
    solvent: [RF, GBSA]
  protocol: absolute-binding
```

**Comment by jchodera** [comment]
This looks great!

**Comment by andrrizzi** [comment]
I think I've finally implemented all the pieces of the pipeline to make this work. All 16 combinations in the protocol terminated successfully. The preliminary results are on the cluster in `/andrrizzi/test_kinase/`.

I've added an md file in #6 with notes from our last meeting about possible combinations and questions we want to answer. I'll close this issue so that we can open one more specific to the next question we want to tackle.
